language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/signature/SignatureHelpUtil.java

```diff
@@ -497,28 +498,38 @@ private static boolean withinTextRange(int position, @Nonnull NonTerminalNode no
         }
         NonTerminalNode nodeAtCursor = context.getNodeAtCursor().get();
         if (nodeAtCursor.kind() == SyntaxKind.FUNCTION_CALL) {
-            NameReferenceNode nameReferenceNode = ((FunctionCallExpressionNode) nodeAtCursor).functionName();
-            String funcName;
-            Predicate<Symbol> symbolPredicate =
-                    symbol -> symbol.kind() == FUNCTION || symbol.kind() == VARIABLE || symbol.kind() == PARAMETER;
-            List<Symbol> filteredContent;
-            if (nameReferenceNode.kind() == SyntaxKind.QUALIFIED_NAME_REFERENCE) {
-                QualifiedNameReferenceNode qNameRef = (QualifiedNameReferenceNode) nameReferenceNode;
-                funcName = (qNameRef).identifier().text();
-                filteredContent = QNameReferenceUtil.getModuleContent(context, qNameRef,
-                                                                      symbolPredicate
-                                                                              .and(symbol -> symbol.getName().orElse("")
-                                                                                      .equals(funcName)));
-            } else {
-                funcName = ((SimpleNameReferenceNode) nameReferenceNode).name().text();
-                List<Symbol> visibleSymbols = context.visibleSymbols(context.getCursorPosition());
-                filteredContent = visibleSymbols.stream()
-                        .filter(symbolPredicate.and(symbol -> symbol.getName().get().equals(funcName)))
-                        .collect(Collectors.toList());
-            }
+            return getFunctionSymbol(((FunctionCallExpressionNode) nodeAtCursor), context);
```

Comment:
I don't think we need to explicitly cast here. In the runtime, this should get mapped to the matching overloaded method.

Reply:
We need the explicit cast here. Otherwise the `getFunctionSymbol(Node nodeAtCursor, SignatureContext context)` will be implemented.